test-suite/js/src/test/scala/org/scalajs/testsuite/utils/JSAssert.scala

```diff
@@ -0,0 +1,21 @@
+package org.scalajs.testsuite.utils
+
+import scala.scalajs.js
+
+import org.junit.Assert._
+
+object JSAssert {
+
+  def assertJSArrayEquals[T](expected: js.Array[T], actual: js.Array[T]): Unit = {
+    assertArrayEquals(expected.map(_.asInstanceOf[AnyRef]).toArray,
+        actual.map(_.asInstanceOf[AnyRef]).toArray)
```

Comment:
I would really like a custom comparison of the arrays, using only `{expected,actual}.{length,apply}`. Our test infrastructure should not rely on `.toArray` working correctly.